sdk/azidentity/example_user_auth_test.go

```diff
@@ -0,0 +1,73 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity_test
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+// This example shows how to authenticate a user with [InteractiveBrowserCredential], enabling persistent
+// caching so that the user doesn't need to authenticate interactively the next time the application runs.
+func Example_userAuthentication() {
+	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
+		// By default, credentials begin interactive authentication whenever necessary. To instead control when
+		// a credential prompts for user interaction, set this option true. The credential will then return
+		// azidentity.ErrAuthenticationRequired instead of prompting for authentication. The application
+		// can then call the credential's Authenticate method when it's convenient to prompt the user.
+		DisableAutomaticAuthentication: true,
+
+		// By default, credentials cache in memory. Set TokenCachePersistenceOptions to enable persistent caching.
+		TokenCachePersistenceOptions: &azidentity.TokenCachePersistenceOptions{
+			// optionally set Name to isolate this credential's cache from other applications
+			Name: "myapp",
+		},
+	})
+	if err != nil {
+		// TODO: handle error
```

Comment:
I would put the error handling in the sample, rather than a TODO.

Reply:
This is the pattern we've adopted in all of our samples.  Makes it clear that if one copy/pastes there's more work to be done in order to properly integrate it.